examples/ensemble/plot_gradient_boosting_categorical.py

```diff
@@ -140,24 +143,14 @@
 # that will natively handle categorical features. This estimator will not treat
 # categorical features as ordered quantities.
 #
-# Since the :class:`~ensemble.HistGradientBoostingRegressor` requires category
-# values to be encoded in `[0, n_unique_categories - 1]`, we still rely on an
-# :class:`~preprocessing.OrdinalEncoder` to pre-process the data.
+# To benefit from this, the categorical features need to be encoded using the
+# pandas categorical dtype which we already did at the beginning of this
+# example with the call to `.astype("category")`.
 #
-# The main difference between this pipeline and the previous one is that in
-# this one, we let the :class:`~ensemble.HistGradientBoostingRegressor` know
-# which features are categorical.
-
-# The ordinal encoder will first output the categorical features, and then the
-# continuous (passed-through) features
-
-hist_native = make_pipeline(
-    ordinal_encoder,
-    HistGradientBoostingRegressor(
-        random_state=42,
-        categorical_features=categorical_columns,
-    ),
-).set_output(transform="pandas")
+# Note that this is equivalent to using the ordinal encoder and then passing
+# the name of the categorical features to the ``categorical_features``
+# constructor parameter of :class:`~ensemble.HistGradientBoostingRegressor`.
```

Comment:
Some users may not want or be able to rely on pandas, so I wonder if we should still leave an example with the `OridnalEncoder` somewhere? Perhaps just as a commented-out snippet here?

Also, in the [User Guide](https://output.circle-artifacts.com/output/job/ff70d45e-4b86-48cc-a0e3-5bd4d686b475/artifacts/0/doc/modules/ensemble.html#categorical-features-support) we currently make a reference to this example, and it probably needs to be updated: 

> The cardinality of each categorical feature should be less than the max_bins parameter, and each categorical feature is expected to be encoded in [0, max_bins - 1]. To that end, it might be useful to pre-process the data with an [OrdinalEncoder](https://output.circle-artifacts.com/output/job/ff70d45e-4b86-48cc-a0e3-5bd4d686b475/artifacts/0/doc/modules/generated/sklearn.preprocessing.OrdinalEncoder.html#sklearn.preprocessing.OrdinalEncoder) as done in [Categorical Feature Support in Gradient Boosting](https://output.circle-artifacts.com/output/job/ff70d45e-4b86-48cc-a0e3-5bd4d686b475/artifacts/0/doc/auto_examples/ensemble/plot_gradient_boosting_categorical.html#sphx-glr-auto-examples-ensemble-plot-gradient-boosting-categorical-py).

Reply:
Good points, thanks.

Reply:
I pushed 75b1afe to address this point.